packages/ui/core-components/src/lib/atoms/inputs/date-range/DateRange.svelte

```diff
@@ -52,30 +52,46 @@
 	}
 
 	const YYYYMMDD = /^\d{4}-\d{2}-\d{2}$/;
-	$: startString =
-		typeof start === 'string' && YYYYMMDD.test(start)
-			? start
-			: start instanceof Date
-				? dateToYYYYMMDD(start)
-				: $query?.[0].start instanceof Date
-					? dateToYYYYMMDD($query?.[0].start)
-					: dateToYYYYMMDD(new Date(0));
-	$: endString =
-		typeof end === 'string' && YYYYMMDD.test(end)
-			? end
-			: end instanceof Date
-				? dateToYYYYMMDD(end)
-				: $query?.[0].end instanceof Date
-					? dateToYYYYMMDD($query?.[0].end)
-					: dateToYYYYMMDD(new Date());
+
+	let startString;
+	$: if (
+		typeof startString !== 'string' ||
+		!YYYYMMDD.test(startString) ||
+		startString === dateToYYYYMMDD(new Date(0))
+	) {
+		startString =
+			typeof start === 'string' && YYYYMMDD.test(start)
+				? start
+				: start instanceof Date
+					? dateToYYYYMMDD(start)
+					: $query?.[0].start instanceof Date
+						? dateToYYYYMMDD($query?.[0].start)
+						: dateToYYYYMMDD(new Date(0));
```

Comment:
These were here before, but if you're up for it - I think these ternaries would benefit from being unwrapped, it is very difficult to see what is happening here

Reply:
Done